core/src/worker/activities/activity_heartbeat_manager.rs

```diff
@@ -74,15 +71,116 @@ pub enum ActivityHeartbeatError {
     /// to heartbeat.
     #[error("Unable to parse activity heartbeat timeout.")]
     InvalidHeartbeatTimeout,
-    /// Core is shutting down and thus new heartbeats are not accepted
-    #[error("New heartbeat requests are not accepted while shutting down")]
-    ShuttingDown,
 }
 
 /// Manages activity heartbeating for a worker. Allows sending new heartbeats or requesting and
 /// awaiting for the shutdown. When shutdown is requested, signal gets sent to all processors, which
 /// allows them to complete gracefully.
 impl ActivityHeartbeatManager {
+    /// Creates a new instance of an activity heartbeat manager and returns a handle to the user,
+    /// which allows to send new heartbeats and initiate the shutdown.
+    /// Returns the manager and a channel that buffers cancellation notifications to be sent to Lang.
+    pub fn new(client: Arc<dyn WorkerClient>) -> (Self, UnboundedReceiver<PendingActivityCancel>) {
```

Comment:
I just moved the constructor from where it was before, there was a duplicate `impl` block.